compute_tools/src/compute.rs

```diff
@@ -109,6 +109,20 @@ impl ComputeState {
             metrics: ComputeMetrics::default(),
         }
     }
+
+    pub fn set_status(&mut self, status: ComputeStatus, state_changed: Option<&Condvar>) {
```

Comment:
Is the state_changed arg ever None?

Reply:
No, not how I've used it.

Reply:
Any preference on removing the option? Or do you anticipate a case where you might want to update the state and not announce it?

Reply:
I went ahead and removed it.